lib/media/track/localmediatrack.js

```diff
@@ -104,11 +131,60 @@ function mixinLocalMediaTrack(AudioOrVideoTrack) {
 
     stop() {
       this._log.info('Stopping');
+      if (this._workaroundWebKitBug1208516Cleanup) {
+        this._workaroundWebKitBug1208516Cleanup();
+        this._workaroundWebKitBug1208516Cleanup = null;
+      }
       this.mediaStreamTrack.stop();
       this._end();
       return this;
     }
   };
 }
 
+function restartWhenInadvertentlyStopped(localMediaTrack) {
+  const { _getUserMedia: getUserMedia, _log: log } = localMediaTrack;
+
+  function shouldReacquireTrack() {
+    return document.visibilityState === 'visible'
+      && (localMediaTrack.mediaStreamTrack.muted || localMediaTrack.isStopped);
+  }
+
+  function reacquireTrack() {
+    const constraints = { audio: false, video: false };
+    constraints[localMediaTrack.mediaStreamTrack.kind] = localMediaTrack.mediaStreamTrack.getConstraints();
+
+    log.info('Re-acquiring the MediaStreamTrack.');
+    log.debug('Constraints:', constraints);
+    return getUserMedia(constraints).then(mediaStream => {
+      return mediaStream.getTracks()[0];
+    });
+  }
+
+  function handleTrackStateChange() {
+    Promise.race([
+      waitForEvent(localMediaTrack.mediaStreamTrack, 'unmute'),
+      waitForSometime(50)
+    ]).then(() => {
+      if (!shouldReacquireTrack()) {
+        return;
+      }
+      reacquireTrack().then(newTrack => {
+        log.info('Re-acquired the MediaStreamTrack.');
+        log.debug('MediaStreamTrack:', newTrack);
+        localMediaTrack._setMediaStreamTrack(newTrack);
```

Comment:
return here is not necessary since this is event  handler. But I see your point of being consisten with return, I will fold this into `shouldReacquireTrack` check.